app/models/exchange_rates/rates_list.rb

```diff
@@ -0,0 +1,45 @@
+module ExchangeRates
+  class RatesList
+    attr_accessor :year,
+                  :month,
+                  :publication_date,
+                  :exchange_rate_files,
+                  :exchange_rates
+
+    def id
+      "#{year}-#{month}-exchange_rate_period"
+    end
+
+    def exchange_rate_file_ids
+      exchange_rate_files.map(&:id)
+    end
+
+    def exchange_rate_ids
+      exchange_rates.map(&:id)
+    end
+
+    class << self
+      def build(month, year)
+        rates_list = new
+        rates_list.year = year
+        rates_list.month = month
+        rates_list.publication_date = exchange_rate_files(month, year).first.publication_date
+        rates_list.exchange_rate_files = exchange_rate_files(month, year)
+        rates_list.exchange_rates = exchange_rates(month, year)
+        rates_list
+      end
+
+      def exchange_rate_files(month, year)
+        files = ExchangeRateCurrencyRate.files_for_year_and_month(month, year)
```

Comment:
You can use ExchangeRateFile.where(month:, year:) here and you can delete the duplicate ExchangeRateFile class in app/models/exchange_rates